src/app/components/Map/Map.tsx

```diff
@@ -17,61 +16,232 @@ if (typeof import.meta.env.VITE_MAPBOX_ACCESSKEY === 'string') {
 export default function MapBox(): JSX.Element {
   const mapContainer = useRef<HTMLDivElement | null>(null)
   const map = useRef<null | Map>(null)
-  const [lng, setLng] = useState<number>(9.96)
-  const [lat, setLat] = useState<number>(53.55)
-  const [zoom, setZoom] = useState<number>(8)
+  const [distance, setDistance] = useState<number>(0)
+  const [location, setLocation] = useState<number[]>([])
+  const [locationName, setLocationName] = useState<string>('')
+  const [location1, setLocation1] = useState<number[]>([])
+  const [location2, setLocation2] = useState<number[]>([])
 
+  // initialize map only once
   useEffect(() => {
-    if (map && map.current) return // initialize map only once
+    if (map && map.current) return
     map.current = new mapboxgl.Map({
       container: mapContainer.current as HTMLElement,
-      style: 'mapbox://styles/mapbox/streets-v11',
-      center: [lng, lat],
-      zoom: zoom,
+      style: 'mapbox://styles/mapbox/outdoors-v11',
+      center: [10, 53.55],
+      zoom: 7,
     })
+    //add MapControl
+    const mapControl = new mapboxgl.NavigationControl()
+    map.current.addControl(mapControl)
 
-    const mapDirections = new MapboxDirections({
+    //initiallize Geocoder and set to Div
+    const geocoder = new MapboxGeocoder({
       accessToken: mapboxgl.accessToken,
     })
-    map.current.addControl(mapDirections, 'top-left')
-  }, [])
+    geocoder.addTo('#locationInput')
 
-  useEffect(() => {
-    map.current?.on('move', () => {
-      if (map.current) {
-        setLng(map.current.getCenter().lng)
-        setLat(map.current.getCenter().lat)
-        setZoom(map.current.getZoom())
-      }
+    geocoder.on('result', function (results) {
+      const location = results.result.center
+      setLocation(location)
+      setLocationName(results.result.text)
     })
-    // wait for map to initialize
   }, [])
 
+  // get Distance from API to [7.43861, 46.95083] (its Bremen) and draw route
+  async function getRoute(start: number[], end: number[]) {
+    const query = await fetch(
+      `https://api.mapbox.com/directions/v5/mapbox/driving/${start[0]},${start[1]};${end[0]},${end[1]}?steps=true&geometries=geojson&access_token=${mapboxgl.accessToken}`,
+      { method: 'GET' }
+    )
+    const json = await query.json()
+    const data = json.routes[0]
+    setDistance(data.distance)
+    const route = data.geometry.coordinates
+
+    const geojson = {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        type: 'LineString',
+        coordinates: route,
+      },
+    }
+
+    const pointData = {
+      type: 'FeatureCollection',
+      features: [
+        {
+          type: 'Feature',
+          properties: {},
+          geometry: {
+            type: 'Point',
+            coordinates: location1,
+          },
+        },
+      ],
+    }
+    map.current
```

Comment:
you could hand over this map object in the function header, then you could define the function somewhere else